addons/mail/static/src/core/common/message_actions.js

```diff
@@ -68,17 +68,46 @@ registerMessageAction("reaction", {
     sequence: 10,
 });
 registerMessageAction("reply-to", {
-    condition: (component) => component.props.message.canReplyTo(component.props.thread),
+    condition: (component) => {
+        const { message, thread } = component.props;
```

Comment:
Avoid destructuring reactive objects like `component.props`: this prevents proper propagation of the actual `this` that is observing the reactive object, therefore can lead to subtle bugs like reactive object changes but it doesn't re-render automatically.